Stratis.Bitcoin/Features/Consensus/ConsensusLoop.cs

```diff
@@ -28,7 +28,8 @@ public ConsensusError Error
 
     public class ConsensusLoop
     {
-        public ConsensusLoop(PowConsensusValidator validator, ConcurrentChain chain, CoinView utxoSet, LookaheadBlockPuller puller, NodeDeployments nodeDeployments, StakeChain stakeChain = null)
+        public ConsensusLoop(PowConsensusValidator validator, ConcurrentChain chain, CoinView utxoSet, LookaheadBlockPuller puller, NodeDeployments nodeDeployments, StakeChain stakeChain = null,
+            FullNode fullNode = null)
```

Comment:
You don't need fullnode all the required classes can all be injected (DateTimeProvider,Settings,Network)